src/test/java/org/broadinstitute/hellbender/tools/dataflow/transforms/InsertSizeMetricsTransformUnitTest.java

```diff
@@ -0,0 +1,329 @@
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import htsjdk.samtools.metrics.Header;
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.metrics.StringHeader;
+import htsjdk.samtools.util.Histogram;
+import htsjdk.samtools.util.TestUtil;
+import org.broadinstitute.hellbender.engine.dataflow.GATKTestPipeline;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.ReadsDataflowSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.metrics.MetricAccumulationLevel;
+import org.broadinstitute.hellbender.tools.dataflow.MetricsFileDataflow;
+import org.broadinstitute.hellbender.tools.picard.analysis.CollectInsertSizeMetrics;
+import org.broadinstitute.hellbender.tools.picard.analysis.InsertSizeMetrics;
+import org.broadinstitute.hellbender.utils.dataflow.DataflowUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.EnumSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+
+
+public final class InsertSizeMetricsTransformUnitTest{
+
+    private static final String SMALL_BAM = "org/broadinstitute/hellbender/tools/picard/analysis/CollectInsertSizeMetrics/insert_size_metrics_test.bam";
+    private static final String METRICS_PATH = "org/broadinstitute/hellbender/metrics/";
+
+    @DataProvider(name = "testFiles")
+    public Object[][] testFiles(){
+        return new Object[][] {
+                {SMALL_BAM, METRICS_PATH + "insertSizeMetricsResultsAllWays.metric", EnumSet.allOf(MetricAccumulationLevel.class)},
+                {SMALL_BAM, METRICS_PATH + "insertSizeMetric_All", EnumSet.of(MetricAccumulationLevel.ALL_READS)},
+                {SMALL_BAM, METRICS_PATH + "insertSizeMetric_Sample", EnumSet.of(MetricAccumulationLevel.SAMPLE)},
+                {SMALL_BAM, METRICS_PATH + "insertSizeMetric_Read_Group", EnumSet.of(MetricAccumulationLevel.READ_GROUP)},
+                {SMALL_BAM, METRICS_PATH + "insertSizeMetric_Library", EnumSet.of(MetricAccumulationLevel.LIBRARY)},
+                {METRICS_PATH + "HiSeq.1mb.1RG.2k_lines.bam", METRICS_PATH + "largeMetricsRun.metrics", EnumSet.allOf(MetricAccumulationLevel.class)}
+        };
+    }
+
+    @Test(groups = "dataflow", dataProvider = "testFiles")
+    public void testInsertSizeMetricsTransformOnFile(String bamName, String expectedMetrics, Set<MetricAccumulationLevel> accumulationLevels) throws IOException {
```

Comment:
There's a pair of essentially identical tests here.  One tests the output against premade files, the other tests the same output against the output of an actual run of the non-dataflow tool.  Both have advantages.  What do people think about testing directly against the existing tool (which will update as we update it) vs testing against premade files?